File: go/application/usecases/inventory_service.go
package usecases

import (
	"taller/application/ports/in"
	"taller/application/ports/out"
	"taller/domain/model"
)

type InventoryService struct {
	inventory out.InventoryRepository
}

func NewInventoryService(inventory out.InventoryRepository) in.InventoryPort {
	return &InventoryService{
		inventory: inventory,
	}
}
func (i *InventoryService) GetCoffeeBeans() ([]model.CoffeeBean, error) {
	beans, err := i.inventory.GetCoffeeBeans()
	if err != nil {
		return nil, err
	}
	return beans, nil
}

File: typescript/src/application/usecases/inventory-service.ts
import { InventoryPort } from "../ports/in/inventory-port";
import { InventoryRepository } from "../ports/out/inventory-repository";
import { CoffeeBean } from "../../domain/model/coffee-bean";

export class InventoryService implements InventoryPort {
  private inventory: InventoryRepository;

  constructor(inventory: InventoryRepository) {
    this.inventory = inventory;
  }

  getCoffeeBeans(): CoffeeBean[] {
    return this.inventory.getCoffeeBeans();
  }
}
